contracts/staking/GoodFundManager.sol

```diff
@@ -300,11 +302,11 @@ contract GoodFundManager is DAOContract {
 
      ) public {
         
-        IGoodDollar token = IGoodDollar(address(avatar.nativeToken()));
+        (uint blockReward,uint startBlock,uint endBlock, bool isBlackListed) = getStakingReward(address(_staking));
```

Comment:
(,,isBlacklisted)
- mintReward should not receive _staking as param. , _staking should be the caller, ie msg.sender
- you need to verify that msg.sender(ie _staking) actually has an entry in stakingRewards ie startBlock<block.number